### DISCUSSION TITLE
Boxing a cached struct

### DISCUSSION THREAD

**Discussion by derolf** [heremaps/gluecodium#1537] [Q&A]
Is there a nicer way to create a reference to a cached struct inside another struct?

Current workaround:

```
@Equatable
@Immutable
struct Big {
...
}

class BigReference {
    @Cached
    property big: Big { get }
}

@Equatable
@Immutable
struct Thing {
  big: BigReference
}
```

Ideally, this would be nice:

```
@Equatable
@Immutable
struct Thing {
  @CachedReference
  big: Big
}
```

The semantics of `@CachedReference big: Big` would be:
- C++: `shared_ptr<Big> big` field in `Thing`
- Java/Swift/Dart: Access to `big` will always yield the same cached object.

**Comment by Hsilgos** [comment]
Sounds like pretty rare case. Currently your approach looks like only possible.
